docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1325 @@
+2024-06-09T19:00:10.696Z|00163|acl_log(ovn_pinctrl0)|INFO|name="ANP:cluster-control:Egress:1", verdict=allow, severity=alert, direction=from-lport: tcp,vlan_tci=0x0000,dl_src=0a:58:0a:f4:02:03,dl_dst=0a:58:0a:f4:02:01,nw_src=10.244.2.3,nw_dst=172.18.0.3,nw_tos=0,nw_ecn=0,nw_ttl=64,nw_frag=no,tp_src=57658,tp_dst=6443,tcp_flags=psh|ack
+2024-06-09T19:00:10.698Z|00164|acl_log(ovn_pinctrl0)|INFO|name="ANP:cluster-control:Egress:1", verdict=allow, severity=alert, direction=from-lport: tcp,vlan_tci=0x0000,dl_src=0a:58:0a:f4:02:03,dl_dst=0a:58:0a:f4:02:01,nw_src=10.244.2.3,nw_dst=172.18.0.3,nw_tos=0,nw_ecn=0,nw_ttl=64,nw_frag=no,tp_src=57658,tp_dst=6443,tcp_flags=ack
+2024-06-09T19:00:11.386Z|00165|acl_log(ovn_pinctrl0)|INFO|name="ANP:cluster-control:Egress:5", verdict=drop, severity=alert, direction=from-lport: icmp,vlan_tci=0x0000,dl_src=0a:58:0a:f4:02:07,dl_dst=0a:58:0a:f4:02:08,nw_src=10.244.2.7,nw_dst=10.244.2.8,nw_tos=0,nw_ecn=0,nw_ttl=64,nw_frag=no,icmp_type=8,icmp_code=0
+```
+
+The same is applicable to debug baseline admin network policy as well
+except for the fact that there is no `pass` in BANP:
+
+```shell
+kubectl annotate banp default k8s.ovn.org/acl-logging='{ "deny": "alert", "allow": "alert" }'
+```
+
+### Ensuring NBDB objects are correctly created
+
+See the details outlined in the OVN constructs section on
+what are the OVN database constructs created for ANPs and BANP
+in a cluster. For example, in order to troubleshoot the
+`cluster-control` ANP at priority 34, we can list the address-sets,
+acls and port-group like shown in the above section. But
+that will list all the acls, address-sets and port-groups
+in your cluster also used by other features. To narrow it
+down to the specific policy in question you can grep for
+the name of the ANP you care about:
+
+* `ovn-nbctl list acl | grep cluster-control` (Do grep with -A, -B, -C to see the full ACL)
+* `ovn-nbctl list address-set | grep anp-name`
+* `ovn-nbctl list port-group | grep anp-name`
+
+This will print out all the ACLs and address-sets and
+portgroups owned by that ANP. In order to troubleshoot
+a specific rule; example Rule5 `default-deny` in egress
+of `cluster-control` ANP:
+
+```yaml
+  - name: "default-deny"
+    action: "Deny"
+    to:
+    - networks:
+      - 0.0.0.0/0
+```
+
+We can start with the knowledge that this rule must be
+translated into one or more ACLs like mentioned in the
+above section. Since there is no L4 match it should be
+translated to a single ACL. Knowing the rule number, we
+can grep for the correct ACL:
+
+`ovn-nbctl list acl | grep cluster-control:Egress:5`
+
+So `<anpName>:<direction>:<ruleNumber>` can be the grep
+pattern to get the right ACL. For this rule it looks like this:
+
+```yaml
+sh-5.2# ovn-nbctl list acl | grep cluster-control:Egress:5 -C 4
```

Comment:
I posted this as a review by accident, but I have had problems in the past, so I deleted it and re-posted as a single comment:

Wouldn't it be cleaner to do this instead?
```
ovn-nbctl list ACL ANP:cluster-control:Egress:5
```

Part of the reason why I am suggesting to avoid `grep` in my recent suggestions is so that we can easily use the format-tweaking parameters of `ovn-nbctl` (e.g. `--format table`), which to me it is often helpful.

In addition, maybe we should have an example with the `<anpName>:<direction>:<ruleNumber>` in the command?

Reply:
Then, I guess it could be:
```
ovn-nbctl find ACL 'external_ids{>=}{"k8s.ovn.org/owner-type"=AdminNetworkPolicy,"k8s.ovn.org/name"=cluster-control,gress-index="5"}'
```

Reply:
agreed that externalIDs is much better, will update it with that

Reply:
done!